lib/util/http/index.ts

```diff
@@ -81,8 +81,11 @@ export class Http<GetOptions = HttpOptions, PostOptions = HttpPostOptions> {
     options.hooks = {
       beforeRedirect: [
         (opts: any): void => {
-          // Check if request has been redirected to Amazon
-          if (opts.search?.includes('X-Amz-Algorithm')) {
+          // Check if request has been redirected to Amazon or an Azure blob (ACR)
+          if (
+            opts.search?.includes('X-Amz-Algorithm') ||
+            opts.hostname?.endsWith('blob.core.windows.net')
```

Comment:
```suggestion
            opts.search?.includes('X-Amz-Algorithm') ||
            opts.hostname?.endsWith('blob.core.windows.net')  // lgtm [js/incomplete-url-substring-sanitization]
```

@etiennetremel Maybe we should check for the token in `search` like amazon instead of `hostname` test

Reply:
It's a bit tricky, Azure has a different behavior. There is nothing in `ops.search` that provide something similar to Amazon. Here is an example:

```
search: '?se=2020-07-01T07%3A44%3A59Z&sig=<sig>&sp=r&sr=b&sv=2016-05-31&regid=<regid>',
```

If you are worried about the host check, I can make it a bit more strict and match both the hostname and the presence of the path `/docker/registry` in the `opts.href`, which look like this:

```
href: 'https://<store>.blob.core.windows.net/<some id>//docker/registry/v2/blobs/sha256/53/<some id>/data?se=2020-07-01T07%3A44%3A59Z&sig=<sig>&sp=r&sr=b&sv=2016-05-31&regid=<regid>',
```

What do you think?

Reply:
ok, please make it more strict. you can add a module scoped function to extract the azure check, so we don't have a huge if condition.